File: Cap 07/7.3 - Metodos de Pesquisa de Elementos/7.3.1  - Pesquisa Sequencial/Alfanumerica/pesqseqalfa.kt
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 7 - Topico 7.3
 * 
 * Programa ..: PESQUISA SEQUENCIAL POR NOME
 * 
 */

import java.util.Scanner

fun main() {
  
    var entrada = Scanner(System.`in`)
    var NOME = Array<String>(10) { "" }
    var I: Int
    var PESQ: String
    var RESP: String
    var ACHA: Boolean

    println("PESQUISA SEQUENCIAL DE NOMES\n")

    for (i in 0..9) {
        print("Entre o ${"%2d".format(i + 1)}o. nome: ")
        NOME[i] = entrada.nextLine()
    }

    // *** Inicio do trecho de pesquisa sequencial ***

    RESP = "SIM"
    while (RESP == "SIM") {
        println()
        print("Entre o nome a ser pesquisado: ")
        PESQ = entrada.nextLine()
        I = 0
        ACHA = false
        while (I <= 9 && ACHA == false) {
            if (PESQ == NOME[I]) {
                ACHA = true
            } else {
                I = I + 1
            }
        }
        if (ACHA == true) {
            println()
            println("$PESQ foi localizado na posicao ${I + 1}")
        } else {
            println()
            println("$PESQ nao foi localizado")
        }
        println()
        print("Deseja continuar? (SIM/NAO): ")
        RESP = entrada.nextLine().uppercase()
    }

    // *** Fim do trecho de pesquisa sequencial ***
	
}

File: Cap 07/7.3 - Metodos de Pesquisa de Elementos/7.3.1  - Pesquisa Sequencial/Alfanumerica/pesqseqalfa.py
'''
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 7 - Topico 7.3
 *
 * Programa ..: PESQUISA SEQUENCIAL POR NOME
'''

NOME = [""] * 10
I = 0
PESQ = ""
RESP = ""
ACHA = False

print("PESQUISA SEQUENCIAL DE NOMES")
print()

for I in range(10): # ou use: range(1, 11, 1)
  print("Entre o {:2d}o. nome: ".format(I + 1), end="")
  NOME[I] = input()

# *** Inicio do trecho de pesquisa sequencial ***

RESP = "SIM"
while (RESP == "SIM"):
  print()
  PESQ = input("Entre o nome a ser pesquisado: ")
  I = 1
  ACHA = False
  while (I <= 10 and ACHA == False):
    if (PESQ == NOME[I - 1]):
      ACHA = True
    else:
      I = I + 1
  if (ACHA == True):
    print()
    print("{0} foi localizado na posicao {1}".format(PESQ, I))
  else:
    print()
    print("{0} nao foi localizado".format(PESQ))
  print()
  RESP = input("Deseja continuar? (SIM/NAO): ").upper()

# *** Fim do trecho de pesquisa sequencial ***
